test/widgets/main_test.dart

```diff
@@ -0,0 +1,179 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:motis_mitfahr_app/main.dart';
+import 'package:motis_mitfahr_app/main_app.dart';
+import 'package:motis_mitfahr_app/util/locale_manager.dart';
+import 'package:motis_mitfahr_app/util/supabase.dart';
+import 'package:motis_mitfahr_app/util/theme_manager.dart';
+import 'package:motis_mitfahr_app/welcome/pages/reset_password_page.dart';
+import 'package:motis_mitfahr_app/welcome/pages/welcome_page.dart';
+
+import '../util/factories/profile_factory.dart';
+import '../util/mock_server.dart';
+import '../util/pump_material.dart';
+import '../util/request_processor.dart';
+import '../util/request_processor.mocks.dart';
+
+void main() {
+  final MockRequestProcessor processor = MockRequestProcessor();
+
+  setUpAll(() async {
+    MockServer.setProcessor(processor);
+  });
+
+  group('AppWrapper', () {
+    testWidgets('It creates the app', (WidgetTester tester) async {
+      await tester.pumpWidget(const AppWrapper());
+
+      expect(find.byType(AppWrapper), findsOneWidget);
+
+      // ignore: invalid_use_of_protected_member
+      expect(localeManager.hasListeners, isTrue);
+      localeManager.setCurrentLocale(const Locale('de'));
+      // ignore: invalid_use_of_protected_member
+      expect(themeManager.hasListeners, isTrue);
+      themeManager.setTheme(ThemeMode.dark);
+
+      final Finder materialApp = find.byType(MaterialApp);
+      expect(materialApp, findsOneWidget);
+      final MaterialApp app = materialApp.evaluate().first.widget as MaterialApp;
+      expect(app.home.runtimeType, AuthApp);
+      expect(app.theme, themeManager.lightTheme);
```

Comment:
why is this ligthTheme? i thought you set the theme to dark in line 35.

Reply:
"theme" just means "lightTheme". "themeMode" would be "currentTheme".
No idea why the naming is so bad. I changed this line to check the dark theme to be a bit more explicit.

I would have loved to use "Theme.of(context)" to get the current theme of the app, but that did not work for some reason.